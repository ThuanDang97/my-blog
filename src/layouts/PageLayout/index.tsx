import Head from 'next/head'
import { lazy, ReactNode, Suspense } from 'react'

// Layouts
const Footer = lazy(() => import('@layouts/Footer'))
export interface IPageLayout {
  children: ReactNode
}

const PageLayout = ({ children }: IPageLayout) => {
  return (
    <>
      <Head>
        <title>Code your life in an elegant way</title>
      </Head>
      {children}
      <Suspense fallback={<div>loading.....</div>}>
        <Footer />
      </Suspense>
    </>
  )
}

export default PageLayout
