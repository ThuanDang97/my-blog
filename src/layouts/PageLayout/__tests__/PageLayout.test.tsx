import { render } from '@utils/testUtils'
import PageLayout, { IPageLayout } from '..'

describe('Page Layouts', () => {
  const mockProps: IPageLayout = {
    children: 'Content',
  }
  it('should render Page layouts', () => {
    const { container } = render(<PageLayout {...mockProps} />)

    expect(container).toMatchSnapshot()
  })
})
