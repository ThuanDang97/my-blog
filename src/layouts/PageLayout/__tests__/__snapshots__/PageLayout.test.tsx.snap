// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page Layouts should render Page layouts 1`] = `
<div>
  Content
  <div>
    loading.....
  </div>
</div>
`;
